Keep `xen-pciback.hide` when upgrading

benjamreis opened the pull request:
Get `xen-pciback.hide` from a host's kernel's args 
Give it to the bootloader after the upgrade

Signed-off-by: BenjiReis <benjamin.reis@vates.fr>

stormi commented on diskutil.py:
```diff
@@ -168,6 +168,10 @@ def getDiskList():
+            if os.path.exists(raid_device):
+                util.runCmd2(['mdadm', '--stop', '--force', raid_device])
+                for dev in members:
+                    util.runCmd2(['dd', 'if=/dev/zero', 'of=%s' % dev, 'bs=512', 'count=1'])
```
From what I understand, we zero out the 512 first bytes of each partition? What about mdadm superblocks that would be at the end of the partitions? Someone suggested to write zeroes there too, should we?

nraynaud replied:
it's the first 512 bits of each disk, that destroys their partition too.

stormi replied:
Ah, I thought we were going to iterate over the partitions in case they were part of a raid array (instead of the whole disk itself), to avoid having the mdadm superblock "resurrect" by chance if the new partition scheme is similar to the old one. But maybe destroying the partition table is enough...

nraynaud replied:
good point, let me think.
basically, I was trying not to ever parse the existing RAID configuration, and only decide things based on new information. Maybe it's wrong.

stormi replied:
What I meant is not to parse the existing RAID configuration but rather to read the partition table, and for each partition found, destroy the first and last bytes of the partition just in case there would be a superblock there.
Then maybe also destroy the partition table itself like you did and also probably the end of the disk in case there would be a superblock for a disk-wide RAID.

stormi replied:
Another thought (from the discussion in the issue), it is possible that mdadm would automatically detect a RAID array and start it, which means that we'd need to stop it before wiping things, to avoid errors, like they did in https://github.com/xcp-ng/xcp/issues/107#issuecomment-446467979

nraynaud replied:
yes, that's the parsing issue, we have to look at all the RAID in the system, parse their configuration, and only stop those whose array include one of the disks we are about to destroy.

benjamreis committed:
Keep `xen-pciback.hide` when upgrading

Get `xen-pciback.hide` from a host's kernel's args
Give it to the bootloader after the upgrade

Signed-off-by: BenjiReis <benjamin.reis@vates.fr>

ydirson commented on product.py:
```diff
@@ -381,6 +381,15 @@ def fetchIfaceInfoFromNetworkdbAsDict(bridge, iface=None):
+            kernel_args = boot_config.menu[boot_config.default].getKernelArgs()
+
+            #   - xen-pciback.hide
+            pciback = next((x for x in kernel_args if x.startswith('xen-pciback.hide=')), None)
+            if pciback:
+                results['host-config']['xen-pciback.hide'] = pciback
+
```
This only extracts a single `xen-pciback.hide` param, it may make sense to preserve all of them ?

benjamreis replied:
the kernel modules just get a single pointer to the parameter

stormi commented on product.py:
```diff
@@ -381,6 +381,15 @@ def fetchIfaceInfoFromNetworkdbAsDict(bridge, iface=None):
             if sched_gran:
                 results['host-config']['sched-gran'] = sched_gran
+
+            # Subset of hypervisor arguments
```
```suggestion
            # Subset of dom0 kernel arguments
```

benjamreis committed:
Update product.py

Co-authored-by: Samuel Verschelde <stormi-xcp@ylix.fr>
Signed-off-by: BenjiReis <benjamin.reis@vates.fr>

stormi reviewed (approved):
This is a candidate for upstreaming (with possibly a better commit message that explains the use case - the *why*).

ydirson reviewed (approved):
lgtm

stormi commented:
Next stage : upstreaming. Automated tests will be added a bit later when we resume work on them.

benjamreis commented:
https://github.com/xcp-ng/xcp/issues/459
